src/api/java/com/minecolonies/api/colony/requestsystem/request/IRequestFactory.java

```diff
@@ -0,0 +1,83 @@
+package com.minecolonies.api.colony.requestsystem.request;
+
+import com.minecolonies.api.colony.requestsystem.RequestState;
+import com.minecolonies.api.colony.requestsystem.factory.IFactory;
+import com.minecolonies.api.colony.requestsystem.requester.IRequester;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Marker interface used to specify a factory for requests.
+ * Restricts the output type of the general factory interface to IRequest
+ *
+ * @param <T> The type of request.
+ * @param <R> The request type.
+ */
+public interface IRequestFactory<T, R extends IRequest<T>> extends IFactory<T, R>
```

Comment:
This class is not supposed to be in this PR yet.
Just the Factories part and the IToken part (and there respective implementations)